server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -906,6 +922,43 @@ static Settings aggregateIndexSettings(
         return indexSettings;
     }
 
+    /**
+     * Updates index settings to enable remote store by default based on cluster level settings
+     * @param settingsBuilder index settings builder to be updated with relevant settings
+     * @param requestSettings settings passed in during index create request
+     * @param clusterSettings cluster level settings
+     */
+    private static void updateRemoteStoreSettings(Settings.Builder settingsBuilder, Settings requestSettings, Settings clusterSettings) {
+        if (FeatureFlags.isEnabled(FeatureFlags.REMOTE_STORE)
+            && FeatureFlags.isEnabled(FeatureFlags.REPLICATION_TYPE)
```

Comment:
We need to move these settings under `FEATURE_FLAGGED_CLUSTER_SETTINGS` . In that case, we won't need to put these checks here and user won't be able be set it w/o enabling the feature flags first.

Reply:
thanks for pointing this out. i'll update

Reply:
in our case its two features we are dependent on. FEATURE_FLAGGED_CLUSTER_SETTINGS only supports adding settings which are dependent on single feature